src/Symfony/Component/HttpKernel/HttpCache/ResponseCacheStrategy.php

```diff
@@ -72,19 +90,14 @@ public function update(Response $response)
             $response->setLastModified(null);
         }
 
-        if (!$response->isFresh()) {
-            $this->cacheable = false;
-        }
+        $this->add($response);
 
         if (!$this->cacheable) {
-            $response->headers->set('Cache-Control', 'no-cache, must-revalidate');
+            $response->headers->set('Cache-Control', implode(', ', array_keys(array_filter($this->cacheDirectives))));
```

Comment:
what about the max age allowing client-side caching ?

Reply:
let's say we don't care, we don't have to be *that* smart?

Reply:
well, allowing private caching without telling the browser how long it can cache it is broken: it won't allow caching in well-behaving browsers (and may allow infinite caching if a crappy browser treats the absence of age on the response as an invitation to cache forever instead of for 0s).

Reply:
client side caching can have 2 modes: expiration-based (caching for a timestamp) and validation-based (asking whether the last-modified or the etag they have is still the right one).
When using ESI or SSI (which this code is about), we don't send an Etag or LastModified (which this class is handling) because we cannot build them properly when the final response is built based on embedded responses. So this only leaves the expiration-based caching client-side. If you don't care about it, close your PR as it brings nothing.

Reply:
I'm about to create unit tests to validate that case so we can implement the required checks.

> we don't send an Etag or LastModified (which this class is handling) because we cannot build them properly when the final response is built based on embedded responses.

Actually we *could* do that by generating an ETag based on the complete response. Something like a hash of the whole content. But I think that's for another PR if we want to add that at all.

Reply:
see https://github.com/nicolas-grekas/symfony/pull/13

Reply:
not helpful, as we could still not use Etag validation of the different sub-requests as we would not have Etags for each of them (unless we build the final Etag as a reversible combination of info about each response), and so we could not perform Etag validation in the controllers generating each subresponse.